apps/docs/src/components/IconPage.astro

```diff
@@ -0,0 +1,19 @@
+---
+import { Icons } from './Icons/Icons';
+import { parseComponentWithoutAstro } from '@selleo/core';
+import { Icon } from '@selleo/core/src/Icon';
+
+const { iconsData } = Astro.props;
+
+const iconsList = iconsData.map(({ name }: { name: string }) => {
+  const { HTMLString, ReactString } = parseComponentWithoutAstro(name, Icon);
+
```

Comment:
can you please restructure this function to be more flexible?
so in the first argument it would the take component (Icon) and the second argument will be object of props